sql: TestTelemetryLoggingDataDriven failed

sql.TestTelemetryLoggingDataDriven [failed](https://mesolite.cluster.engflow.com/invocations/default/d0821d5b-2646-4332-84ed-6bf442a986c7?testReportRun=1&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [516a17d899fc81ad62f8baf16dc7ef23a8a9360c](https://github.com/cockroachdb/cockroach/commits/516a17d899fc81ad62f8baf16dc7ef23a8a9360c):


```
    test_server_shim.go:637: DRPC is enabled (via testserver factory defaults)
--- FAIL: TestTelemetryLoggingDataDriven/force_logging (3.56s)
=== RUN   TestTelemetryLoggingDataDriven/force_logging/txn_mode_tracing_on
    datadriven.go:263: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/force_logging:4:
          exec-sql
          SET CLUSTER SETTING sql.telemetry.transaction_sampling.max_event_frequency = 10;
          ----
    datadriven.go:263: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/force_logging:8:
          exec-sql
          SET CLUSTER SETTING sql.telemetry.transaction_sampling.statement_events_per_transaction.max = 100;
          ----
    datadriven.go:263: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/force_logging:12:
          exec-sql
          SET CLUSTER SETTING sql.telemetry.query_sampling.mode = "transaction";
          ----
    datadriven.go:263: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/force_logging:16:
          exec-sql
          SET CLUSTER SETTING sql.telemetry.query_sampling.enabled = true;
          ----
    datadriven.go:263: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/force_logging:20:
          exec-sql
          CREATE TABLE t()
          ----
    datadriven.go:263: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/force_logging:24:
          spy-sql unixSecs=0.1
          SET TRACING = ON;
          ----
    datadriven.go:263: 
          /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/force_logging:28:
          spy-sql unixSecs=0.1 useRealTracing=true
          SELECT * FROM t LIMIT 1;
        output didn't match expected:
          @@ -1,9 +1,9 @@
           {
           	"ApplicationName": "telemetry-logging-datadriven",
           	"Database": "defaultdb",
          -	"Distribution": "local",
          +	"Distribution": "full",
           	"EventType": "sampled_query",
           	"PlanGist": "AgHQAQIAAAAAAg==",
           	"ScanCount": 1,
           	"Statement": "SELECT * FROM \"\".\"\".t LIMIT ‹1›",
           	"StatementFingerprintID": "7201809505509488600",
--- FAIL: TestTelemetryLoggingDataDriven/force_logging/txn_mode_tracing_on (0.04s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetryLoggingDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63630

sql.TestTelemetryLoggingDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21337645?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21337645?buildTab=artifacts#/) on master @ [516a17d899fc81ad62f8baf16dc7ef23a8a9360c](https://github.com/cockroachdb/cockroach/commits/516a17d899fc81ad62f8baf16dc7ef23a8a9360c):


```
          	"OutputRowsEstimate": 1,
          	"PlanGist": "AgICBAYE",
          	"SkippedQueries": 1,
          	"Statement": "SELECT ‹1›, ‹2›",
          	"StatementFingerprintID": "11744053289039399750",
          	"StmtPosInTxn": 1,
          	"Tag": "SELECT",
          	"User": "root"
          }
          {
          	"ApplicationName": "telemetry-logging-datadriven",
          	"Committed": true,
          	"EventType": "sampled_transaction",
          	"NumRows": 1,
          	"RowsRead": 0,
          	"RowsWritten": 0,
          	"SkippedTransactions": 1,
          	"StatementFingerprintIDs": [
          		"11744053289039399750"
          	],
          	"TransactionFingerprintID": "979813787843574937",
          	"User": "root"
          }
    datadriven.go:263: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/17712/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:195:
          spy-sql unixSecs=1.05
          SELECT * FROM t LIMIT 1
          ----
    datadriven.go:263: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/17712/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:200:
          spy-sql unixSecs=1.08
          SELECT * FROM t LIMIT 1
          ----
    datadriven.go:263: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/17712/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:205:
          spy-sql unixSecs=1.1
          SELECT * FROM t LIMIT 2
        output didn't match expected:
          @@ -1,9 +1,9 @@
           {
           	"ApplicationName": "telemetry-logging-datadriven",
           	"Database": "defaultdb",
          -	"Distribution": "local",
          +	"Distribution": "full",
           	"EventType": "sampled_query",
           	"PlanGist": "AgHQAQQAAAAAAg==",
           	"ScanCount": 1,
           	"SkippedQueries": 2,
           	"Statement": "SELECT * FROM \"\".\"\".t LIMIT ‹2›",
--- FAIL: TestTelemetryLoggingDataDriven/transaction_mode/transaction_mode_events (0.16s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetryLoggingDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestTelemetryLoggingDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21337748?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21337748?buildTab=artifacts#/) on master @ [9a4e05584583e85915a5f865e1aa05d98b46889b](https://github.com/cockroachdb/cockroach/commits/9a4e05584583e85915a5f865e1aa05d98b46889b):


```
          	"OutputRowsEstimate": 1,
          	"PlanGist": "AgICBAYE",
          	"SkippedQueries": 1,
          	"Statement": "SELECT ‹1›, ‹2›",
          	"StatementFingerprintID": "11744053289039399750",
          	"StmtPosInTxn": 1,
          	"Tag": "SELECT",
          	"User": "root"
          }
          {
          	"ApplicationName": "telemetry-logging-datadriven",
          	"Committed": true,
          	"EventType": "sampled_transaction",
          	"NumRows": 1,
          	"RowsRead": 0,
          	"RowsWritten": 0,
          	"SkippedTransactions": 1,
          	"StatementFingerprintIDs": [
          		"11744053289039399750"
          	],
          	"TransactionFingerprintID": "979813787843574937",
          	"User": "root"
          }
    datadriven.go:263: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19048/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:195:
          spy-sql unixSecs=1.05
          SELECT * FROM t LIMIT 1
          ----
    datadriven.go:263: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19048/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:200:
          spy-sql unixSecs=1.08
          SELECT * FROM t LIMIT 1
          ----
    datadriven.go:263: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19048/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:205:
          spy-sql unixSecs=1.1
          SELECT * FROM t LIMIT 2
        output didn't match expected:
          @@ -1,9 +1,9 @@
           {
           	"ApplicationName": "telemetry-logging-datadriven",
           	"Database": "defaultdb",
          -	"Distribution": "local",
          +	"Distribution": "full",
           	"EventType": "sampled_query",
           	"PlanGist": "AgHQAQQAAAAAAg==",
           	"ScanCount": 1,
           	"SkippedQueries": 2,
           	"Statement": "SELECT * FROM \"\".\"\".t LIMIT ‹2›",
--- FAIL: TestTelemetryLoggingDataDriven/transaction_mode/transaction_mode_events (4.05s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetryLoggingDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestTelemetryLoggingDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21337989?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21337989?buildTab=artifacts#/) on master @ [611704755ec13a9c25732dfa55bc79e920ccc294](https://github.com/cockroachdb/cockroach/commits/611704755ec13a9c25732dfa55bc79e920ccc294):


```
          	"OutputRowsEstimate": 1,
          	"PlanGist": "AgICBAYE",
          	"SkippedQueries": 1,
          	"Statement": "SELECT ‹1›, ‹2›",
          	"StatementFingerprintID": "11744053289039399750",
          	"StmtPosInTxn": 1,
          	"Tag": "SELECT",
          	"User": "root"
          }
          {
          	"ApplicationName": "telemetry-logging-datadriven",
          	"Committed": true,
          	"EventType": "sampled_transaction",
          	"NumRows": 1,
          	"RowsRead": 0,
          	"RowsWritten": 0,
          	"SkippedTransactions": 1,
          	"StatementFingerprintIDs": [
          		"11744053289039399750"
          	],
          	"TransactionFingerprintID": "979813787843574937",
          	"User": "root"
          }
    datadriven.go:263: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19138/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:195:
          spy-sql unixSecs=1.05
          SELECT * FROM t LIMIT 1
          ----
    datadriven.go:263: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19138/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:200:
          spy-sql unixSecs=1.08
          SELECT * FROM t LIMIT 1
          ----
    datadriven.go:263: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19138/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:205:
          spy-sql unixSecs=1.1
          SELECT * FROM t LIMIT 2
        output didn't match expected:
          @@ -1,9 +1,9 @@
           {
           	"ApplicationName": "telemetry-logging-datadriven",
           	"Database": "defaultdb",
          -	"Distribution": "local",
          +	"Distribution": "full",
           	"EventType": "sampled_query",
           	"PlanGist": "AgHQAQQAAAAAAg==",
           	"ScanCount": 1,
           	"SkippedQueries": 2,
           	"Statement": "SELECT * FROM \"\".\"\".t LIMIT ‹2›",
--- FAIL: TestTelemetryLoggingDataDriven/transaction_mode/transaction_mode_events (0.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetryLoggingDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestTelemetryLoggingDataDriven [failed](https://mesolite.cluster.engflow.com/invocations/default/6e10c8e3-604d-453e-9a8f-53f3397f72d2?testReportRun=4&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [d7a4000609e45d58b8fcb925759a760bfd1e0506](https://github.com/cockroachdb/cockroach/commits/d7a4000609e45d58b8fcb925759a760bfd1e0506):


```
          	"OutputRowsEstimate": 1,
          	"PlanGist": "AgICBAYE",
          	"SkippedQueries": 1,
          	"Statement": "SELECT ‹1›, ‹2›",
          	"StatementFingerprintID": "11744053289039399750",
          	"StmtPosInTxn": 1,
          	"Tag": "SELECT",
          	"User": "root"
          }
          {
          	"ApplicationName": "telemetry-logging-datadriven",
          	"Committed": true,
          	"EventType": "sampled_transaction",
          	"NumRows": 1,
          	"RowsRead": 0,
          	"RowsWritten": 0,
          	"SkippedTransactions": 1,
          	"StatementFingerprintIDs": [
          		"11744053289039399750"
          	],
          	"TransactionFingerprintID": "979813787843574937",
          	"User": "root"
          }
    datadriven.go:263: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:195:
          spy-sql unixSecs=1.05
          SELECT * FROM t LIMIT 1
          ----
    datadriven.go:263: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:200:
          spy-sql unixSecs=1.08
          SELECT * FROM t LIMIT 1
          ----
    datadriven.go:263: 
          /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:205:
          spy-sql unixSecs=1.1
          SELECT * FROM t LIMIT 2
        output didn't match expected:
          @@ -1,9 +1,9 @@
           {
           	"ApplicationName": "telemetry-logging-datadriven",
           	"Database": "defaultdb",
          -	"Distribution": "local",
          +	"Distribution": "full",
           	"EventType": "sampled_query",
           	"PlanGist": "AgHQAQQAAAAAAg==",
           	"ScanCount": 1,
           	"SkippedQueries": 2,
           	"Statement": "SELECT * FROM \"\".\"\".t LIMIT ‹2›",
--- FAIL: TestTelemetryLoggingDataDriven/transaction_mode/transaction_mode_events (0.11s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetryLoggingDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestTelemetryLoggingDataDriven [failed](https://mesolite.cluster.engflow.com/invocations/default/a1c79cd5-dbea-4d2a-9fa5-0fb96276c0d4?testReportRun=2&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [1a34c229dfa74568ba33af51109dc8c86b297264](https://github.com/cockroachdb/cockroach/commits/1a34c229dfa74568ba33af51109dc8c86b297264):


```
          	"OutputRowsEstimate": 1,
          	"PlanGist": "AgICBAYE",
          	"SkippedQueries": 1,
          	"Statement": "SELECT ‹1›, ‹2›",
          	"StatementFingerprintID": "11744053289039399750",
          	"StmtPosInTxn": 1,
          	"Tag": "SELECT",
          	"User": "root"
          }
          {
          	"ApplicationName": "telemetry-logging-datadriven",
          	"Committed": true,
          	"EventType": "sampled_transaction",
          	"NumRows": 1,
          	"RowsRead": 0,
          	"RowsWritten": 0,
          	"SkippedTransactions": 1,
          	"StatementFingerprintIDs": [
          		"11744053289039399750"
          	],
          	"TransactionFingerprintID": "979813787843574937",
          	"User": "root"
          }
    datadriven.go:263: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:195:
          spy-sql unixSecs=1.05
          SELECT * FROM t LIMIT 1
          ----
    datadriven.go:263: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:200:
          spy-sql unixSecs=1.08
          SELECT * FROM t LIMIT 1
          ----
    datadriven.go:263: 
          /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:205:
          spy-sql unixSecs=1.1
          SELECT * FROM t LIMIT 2
        output didn't match expected:
          @@ -1,9 +1,9 @@
           {
           	"ApplicationName": "telemetry-logging-datadriven",
           	"Database": "defaultdb",
          -	"Distribution": "local",
          +	"Distribution": "full",
           	"EventType": "sampled_query",
           	"PlanGist": "AgHQAQQAAAAAAg==",
           	"ScanCount": 1,
           	"SkippedQueries": 2,
           	"Statement": "SELECT * FROM \"\".\"\".t LIMIT ‹2›",
--- FAIL: TestTelemetryLoggingDataDriven/transaction_mode/transaction_mode_events (0.78s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetryLoggingDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestTelemetryLoggingDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21339524?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21339524?buildTab=artifacts#/) on master @ [88ba9beefa2e1b37e72ffedb0af24edaabb68cb4](https://github.com/cockroachdb/cockroach/commits/88ba9beefa2e1b37e72ffedb0af24edaabb68cb4):


```
          	"OutputRowsEstimate": 1,
          	"PlanGist": "AgICBAYE",
          	"SkippedQueries": 1,
          	"Statement": "SELECT ‹1›, ‹2›",
          	"StatementFingerprintID": "11744053289039399750",
          	"StmtPosInTxn": 1,
          	"Tag": "SELECT",
          	"User": "root"
          }
          {
          	"ApplicationName": "telemetry-logging-datadriven",
          	"Committed": true,
          	"EventType": "sampled_transaction",
          	"NumRows": 1,
          	"RowsRead": 0,
          	"RowsWritten": 0,
          	"SkippedTransactions": 1,
          	"StatementFingerprintIDs": [
          		"11744053289039399750"
          	],
          	"TransactionFingerprintID": "979813787843574937",
          	"User": "root"
          }
    datadriven.go:263: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18411/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:195:
          spy-sql unixSecs=1.05
          SELECT * FROM t LIMIT 1
          ----
    datadriven.go:263: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18411/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:200:
          spy-sql unixSecs=1.08
          SELECT * FROM t LIMIT 1
          ----
    datadriven.go:263: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18411/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/testdata/telemetry_logging/logging/transaction_mode:205:
          spy-sql unixSecs=1.1
          SELECT * FROM t LIMIT 2
        output didn't match expected:
          @@ -1,9 +1,9 @@
           {
           	"ApplicationName": "telemetry-logging-datadriven",
           	"Database": "defaultdb",
          -	"Distribution": "local",
          +	"Distribution": "full",
           	"EventType": "sampled_query",
           	"PlanGist": "AgHQAQQAAAAAAg==",
           	"ScanCount": 1,
           	"SkippedQueries": 2,
           	"Statement": "SELECT * FROM \"\".\"\".t LIMIT ‹2›",
--- FAIL: TestTelemetryLoggingDataDriven/transaction_mode/transaction_mode_events (0.33s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetryLoggingDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/telemetry_datadriven_test.go
+++ b/pkg/sql/telemetry_datadriven_test.go
@@ -101,7 +101,12 @@ func TestTelemetryLoggingDataDriven(t *testing.T) {
 			getTimeNow:       st.TimeNow,
 			getTracingStatus: sts.TracingStatus,
 		}
-		tc := serverutils.StartCluster(t, 3, base.TestClusterArgs{
+		// NB: use a single-node cluster so that the leaseholder for any
+		// user table is always the gateway. With multiple nodes, the
+		// leaseholder can land on a non-gateway node and the physical
+		// planner will distribute the flow, making the Distribution field
+		// in telemetry log events nondeterministic. See #169735.
+		tc := serverutils.StartCluster(t, 1, base.TestClusterArgs{
 			ServerArgs: base.TestServerArgs{
 				Knobs: base.TestingKnobs{
 					SQLExecutor: &ExecutorTestingKnobs{

